internal/v4/relations_test.go

```diff
@@ -766,6 +768,53 @@ func (s *RelationsSuite) TestMetaBundlesContaining(c *gc.C) {
 	}
 }
 
+func (s *RelationsSuite) TestMetaBundlesContainingBundleACL(c *gc.C) {
+	// Add the bundles used for testing to the database.
+	for id, b := range metaBundlesContainingBundles {
+		url := mustParseResolvedURL(id)
+		// The blob related info are not used in these tests.
+		// The charm-bundle relations are retrieved from the entities
+		// collection, without accessing the blob store.
+		err := s.store.AddBundle(b, charmstore.AddParams{
+			URL:      url,
+			BlobName: "blobName",
+			BlobHash: fakeBlobHash,
+			BlobSize: fakeBlobSize,
+		})
+		c.Assert(err, gc.IsNil)
+		if url.URL.Name == "useless" {
+			err = s.store.SetPerms(&url.URL, "read", url.URL.User)
+			c.Assert(err, gc.IsNil)
+			continue
+		}
+		err = s.store.SetPerms(&url.URL, "read", params.Everyone, url.URL.User)
+		c.Assert(err, gc.IsNil)
+	}
+	rurl := mustParseResolvedURL("42 ~charmers/utopic/wordpress-42")
+	// Add the charm we need bundle info on to the database.
+	err := s.store.AddCharm(&relationTestingCharm{}, charmstore.AddParams{
+		URL:      rurl,
+		BlobName: "blobName",
+		BlobHash: fakeBlobHash,
+		BlobSize: fakeBlobSize,
+	})
+	c.Assert(err, gc.IsNil)
+	err = s.store.SetPerms(&rurl.URL, "read", params.Everyone, rurl.URL.User)
+	c.Assert(err, gc.IsNil)
+
+	// Perform the request and ensure the response is what we expect.
```

Comment:
Perhaps suggest that we expect the useless bundle to not be included in the results.